python/desc/imsim/camera_readout.py

```diff
@@ -292,29 +329,46 @@ def get_amplifier_hdu(self, amp_name):
             appropriate for the requested sensor segment.
         """
         hdu = fits.ImageHDU(data=self.amp_images[amp_name].getArray().astype(np.int32))
-
+        hdr = hdu.header
+        amp_info = self.camera_info.get_amp_info(amp_name)
         # Copy keywords from eimage primary header.
         with warnings.catch_warnings():
             warnings.simplefilter("ignore")
             for key in self.eimage[0].header.keys():
                 try:
-                    hdu.header[key] = self.eimage[0].header[key]
-                except ValueError as eobj:
+                    hdr[key] = self.eimage[0].header[key]
+                except ValueError:
                     # eimages produced by phosim contain non-ASCII or
                     # non-printable characters resulting in a ValueError.
                     self.logger.warn("ValueError raised while attempting to "
-                                     "read {} from eimage header".format(key))
+                                     "read %s from eimage header", key)
+
+        # Transpose the WCS matrix elements to account for the use of the
+        # Camera Coordinate System in the eimage.  These changes
+        # neglect any implied changes in the SIP coefficients.
```

Comment:
I don't follow this comment about changes in the SIP coefficients.

Reply:
If you look at the header of one of our eimage files, you'll see that there are coefficients for the SIP polynomials that have x- and y-indexes.   In going from CCS to `raw` coordinates, one needs to swap x- and y-directions in the `CD_x_y` array, but I'm not applying that swap to the SIP coefficients.  This explanation seems too complicated to put in a comment, but I think anyone with detailed knowledge of the issues would understand the current comment.